### PR TITLE
Feature/prepare maven central

### PR DESCRIPTION
**Pull request by sappenin**
This PR updates the gradle build file to allow for code signing and artifact deploy to Sonatype, which mirrors out to Maven Central.

Additionally, there are minor formatting and Javadoc changes.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,12 +56,143 @@ dependencies {
...
+        }
+    }
+}
+
 //Maven compatibility task:
 task writePom {
     doLast {
         pom {
             project {
+                name 'ILP Core (Java)'
```
Comment (adrianhopebailie) [inline build.gradle]: s/ILP Core (Java)/Interledger Core/

File: build.gradle [hunk 2]
```diff
@@ -57,12 +56,143 @@ dependencies {
...
+                authentication(userName: hasProperty('ossrhUsername') ? ossrhUsername : '', password: hasProperty('ossrhPassword') ? ossrhPassword : '')
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots") {
+                authentication(userName: hasProperty('ossrhUsername') ? ossrhUsername : '', password: hasProperty('ossrhPassword') ? ossrhPassword : '')
+            }
+
+            pom {
+                project {
+                    name 'ILP Core (Java)'
```
Comment (adrianhopebailie) [inline build.gradle]: s/ILP Core (Java)/Interledger Core/

File: build.gradle (hunk 1 above)
Comment (sappenin) [inline build.gradle]: Done!

**Comment by sappenin** [comment]
This PR is a good first step - it allows me to manually publish SNAPSHOTS and releases, but I'd like to get CircleCI at least publishing SNAPSHOTS when the development branch is built. Unfortunately, I am unable to get gradle to read the environment variables from CircleCI, so that's not yet working -- I plan to spend some more time on that this weekend.

**Comment by sappenin** [comment]
Just pushed one more change for the naming - let me know what you think @adrianhopebailie

**Pull request merged**